### ISSUE TITLE
Remove `analysis` feature

### ISSUE THREAD

**Issue by k-sareen** [mmtk/mmtk-core#904]
We should remove the `analysis` feature given that eBPF support has now been merged. We should not maintain deprecated analysis platforms. eBPF is more powerful.

**Comment by caizixian** [comment]
I'm not sure whether we should remove it. For high frequency events, a statically compiled-in user-space instrumentation has much lower overhead, which matters even if you are just doing offline analysis.

OK, we might need to measure it. If a complete analysis take a day, and using bpftrace is 10% slower, maybe it's acceptable. If a complete analysis takes a hour, and using bpftrace means it's now 10 hours, then probably not.

**Comment by k-sareen** [comment]
I don't think anyone other than me has actually used `analysis`. I'm worried about technical debt. If it doesn't have a big use-case then maybe it doesn't make sense to keep it around. But yes you're right. It would be good to measure overheads

**Comment by caizixian** [comment]
The code footprint seems small enough. If it becomes too difficult to maintain while also not used, then we can consider removal.

**Comment by k-sareen** [comment]
Yeah it uses the same infrastructure as stress so it shouldn't be too onerous to maintain, it's just I felt like it wasn't used/eBPF is more flexible.
